datasources/src/main/elm/sources/APICalls.elm

```diff
@@ -0,0 +1,62 @@
+module APICalls exposing (..)
+
+
+import Model exposing (DataSource, Model)
+import Http exposing (..)
+import Json.Decode
+import JsonDecoder exposing (..)
+import JsonEncoder exposing (encodeDataSource)
+import Messages exposing (..)
+
+getUrl: Model -> String -> String
+getUrl m url =
+  m.contextPath ++ "/secure/api/" ++ url
+
+getDataSources : Model -> Cmd Msg
+getDataSources  model =
+  let
+    req =
+      request
+        { method  = "GET"
+        , headers = []
+        , url     = getUrl model "datasources"
+        , body    = emptyBody
+        , expect  = expectJson GetDataSources (decodeDataList decodeDataSource [ "datasources"] )
+        , timeout = Nothing
+        , tracker = Nothing
+        }
+  in
+    req
+
+saveDataSource : Model -> Bool -> DataSource -> Cmd Msg
+saveDataSource  model isNew datasource =
+  let
+    req =
+      request
+        { method  = if isNew then "PUT" else "POST"
+        , headers = []
+        , url     = getUrl model "datasources"
```

Comment:
when saving an existing datasource, it should be like delete: `datasources/${id}` 
![image](https://github.com/Normation/rudder-plugins/assets/44649/000a3f75-be80-47e3-b37a-b993f63805c3)